sherpa/models/tests/test_regrid.py

```diff
@@ -292,6 +292,17 @@ def tst_runtime_interp(model, requested, interp):
     assert ygot == approx(yexpected)
 
 
+def test_regrid_binaryop():
+    """issue #762, Cannot regrid a composite model (BinaryOpModel)"""
+    import sherpa.astro.ui as ui
+    ui.dataspace1d(-5, 15, 1, dstype=Data1D)
+    b = Box1D()
+    c = Const1D
+    ui.set_source(ui.const1d.c1 + ui.box1d.b1)
+    mdl = ui.get_source()
+    mdl.regrid(np.linspace(-5, 15))
```

Comment:
I would feel more comfortable if the regridded model was used for anything, i.e. call it on some number and check the return value. Right now, that's going to work (the code change is so simple, what could go wrong?), but if we ever change the implementation, it would be better to test that the regridded model actually works.

Reply:
Regarding the code change is so simple, that is the beauty of recursive code.  BTW, if you comment out the three lines of the code change you will find that test will fail but I will add something

Reply:
I know it will. I suggested that to make sure we test enough for the future. I've seen bugs in other software where code was run and tests thus passed, but the numerical result was wrong because the tests did not check the return value. For example, in this case, the following regrid function would also make the test pass:
```
for part in self:
   pass
```
So, I think it's important to actually *use* the regridded function in some way and make sure that something has actually changed as a result of the regrid.